app/background/events/cosmeticFilterEvents.ts

```diff
@@ -0,0 +1,116 @@
+// import windowActions from '../actions/cosmeticFilterActions'
+// import tabActions from '../actions/tabActions'
+
+// import * as cosmeticFilterAPI from '../api/cosmeticFilterAPI'
+// background
+
+// this definitely goes into state
+import cosmeticFilterActions from '../actions/cosmeticFilterActions'
+
+let rule = {
+  host: '',
+  selector: ''
+}
+
+// add context menu
+chrome.runtime.onInstalled.addListener(function () {
+
+  // parent menu
+  chrome.contextMenus.create({
+    title: 'Brave',
+    id: 'brave',
+    contexts: ['all']
+  })
+
+  // block ad child menu
+  chrome.contextMenus.create({
+    title: 'Block ad',
+    id: 'addBlockElement',
+    parentId: 'brave',
+    contexts: ['all']
+  }, () => {
+    console.log('block custom ad sub-context menu created')
+  })
+
+  chrome.contextMenus.create({
+    title: 'Reset site filter settings',
+    id: 'resetSiteFilterSettings',
+    parentId: 'brave',
+    contexts: ['all']
+  }, () => {
+    console.log('reset site cosmetic filter sub-context menu created')
+  })
+
+  chrome.contextMenus.create({
+    title: 'Reset all filter settings',
+    id: 'resetAllFilterSettings',
+    parentId: 'brave',
+    contexts: ['all']
+  }, () => {
+    console.log('log chrome extension storage menu created')
+  })
+
+  // report broken page child menu
+  chrome.contextMenus.create({
+    title: 'Report broken page',
+    id: 'reportBrokenPage',
+    parentId: 'brave',
+    contexts: ['all']
+  }, () => {
+    console.log('report broken page sub-context menu created')
+  })
+
+  // log storage
+  // chrome.contextMenus.create({
+  //   title: 'Log chrome extension storage',
+  //   id: 'logStorage',
+  //   parentId: 'brave',
+  //   contexts: ['all']
+  // }, () => {
+  //   console.log('log chrome extension storage menu created')
+  // })
+})
+
+// contextMenu listener - when triggered, grab latest selector
+chrome.contextMenus.onClicked.addListener(function (info, tab) {
+  switch (info.menuItemId) {
+    case 'addBlockElement':
+      {
+        console.log(rule)
+        console.log('blockElement clicked in cosmeticFilterEvents.ts', rule.host, rule.selector)
+        rule.selector = window.prompt('rule: ', `${rule.selector}`) || ''
+        chrome.tabs.insertCSS({
+          code: `${rule.selector} {display: none;}`
+        })
+        cosmeticFilterActions.siteCosmeticFilterAdded(rule.host, rule.selector)
+        break
+      }
+    case 'resetSiteFilterSettings':
+      {
+        console.log('dispatching cosmeticFilterAction to cosmeticFilterReducer from events')
+        cosmeticFilterActions.siteCosmeticFilterRemoved(rule.host)
+        break
+      }
+    case 'reportBrokenPage':
```

Comment:
not sure if this is needed we are only logging the result inside the action handler